xstatic/pkg/datatables/__init__.py

```diff
@@ -46,7 +46,16 @@
     # your relative path/filenames should usually be without version
     # information, because either the base dir/url is exactly for this
     # version or the mapping will care for accessing this version.
-    ('microsoft', 'http'): 'http://ajax.aspnetcdn.com/ajax/jquery.dataTables/%s' % VERSION,
-    ('microsoft', 'https'): 'https://ajax.aspnetcdn.com/ajax/jquery.dataTables/%s' % VERSION,
+    ('datatables', 'http'): {
+        'jquery.dataTables.css': 'http://cdn.datatables.net/%s/css/jquery.dataTables.css' % VERSION,
+        'jquery.dataTables.min.css': 'http://cdn.datatables.net/%s/css/jquery.dataTables.min.css' % VERSION,
+        'jquery.dataTables.js': 'http://cdn.datatables.net/%s/js/jquery.dataTables.js' % VERSION,
+        'jquery.dataTables.min.js': 'http://cdn.datatables.net/%s/js/jquery.dataTables.min.js' % VERSION,
+    },
```

Comment:
if all the files are under the expected name BELOW a common base dir, you could handle it by a single line per protocol (as you see in the old/removed line for the microsoft cdn).

If xstatic does not find a mapping dict, but just a str, it will just append the filename it is interested in.
The mapping dict is only for resolving more complicated cases.